src/corefx/System.Globalization.Native/localeStringData.cpp

```diff
@@ -240,3 +255,83 @@ extern "C" int32_t GetLocaleInfoString(const UChar* localeName, LocaleStringData
 
 	return UErrorCodeToBool(status);
 }
+
+/*
+Function:
+NormalizeTimePattern
+
+Convert an ICU non-localized time pattern to .NET format
+*/
+void NormalizeTimePattern(const UnicodeString *srcPattern, UnicodeString *destPattern)
+{
+	// An srcPattern example: "h:mm:ss a"
+	// A destPattern example: "h:mm:ss tt"
+	destPattern->remove();
+
+	for (int i = 0; i <= srcPattern->length() - 1; i++)
+	{
+		UChar ch = srcPattern->charAt(i);
+		switch (ch)
+		{
+		case ':':
+		case '.':
+		case 'H':
+		case 'h':
+		case 'm':
+		case 's':
+			destPattern->append(ch);
+			break;
+
+		case chSpace:
```

Comment:
Could we use CAPITALS for these defines and maybe call them SPACE_UCHAR and NBSP_UCHAR?